### PR TITLE
Remove client side recipe allow list

### PR DESCRIPTION
**Pull request by EdwardJTL**
Remove the allow list and retire the recipe-specific onboarding flow.

### How was this tested?
- [x] I tested this manually
- [ ] I added automated tests
- [ ] Existing automated tests are enough to cover my changes

### Manual test cases
tested that regular onboarding still works.

## Issues addressed
#4292

### CODE REVIEW

File: Client.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -63,10 +63,6 @@
 		2028A4C328B8014B00618EFD /* GroupedDataPanelView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2028A4C228B8014B00618EFD /* GroupedDataPanelView.swift */; };
 		2028A4C528B8060400618EFD /* ArchivedTabsGroupedDataModel.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2028A4C428B8060400618EFD /* ArchivedTabsGroupedDataModel.swift */; };
 		2028A4C728B8070000618EFD /* HistoryGroupedDataModel.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2028A4C628B8070000618EFD /* HistoryGroupedDataModel.swift */; };
-		2028A4C828B810EA00618EFD /* GroupedDataPanelView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2028A4C228B8014B00618EFD /* GroupedDataPanelView.swift */; };
```
Comment (darinwf) [inline Client.xcodeproj/project.pbxproj:66]: Unexpected changes?

**Review by darinwf — APPROVED** [review]

File: Client.xcodeproj/project.pbxproj (hunk 1 above)
Comment (EdwardJTL) [reply Client.xcodeproj/project.pbxproj:66]: Ah probably came from the merge...

**Review by darinwf — APPROVED** [review]

**Comment by EdwardJTL** [comment]
waiting for BE to fix their allow list before shipping this...

**Comment by EdwardJTL** [comment]
BE change merged. waiting for shipping to prod

**Comment by EdwardJTL** [comment]
it's fine if we dont cherry pick this. we will just have duplicate guards in BE and client

**Pull request merged**